10-methods.slide

```diff
@@ -0,0 +1,69 @@
+Methods
+22 Feb 2017
+Tags: edmonton-go, golang, workshop, workshop-one, functions, methods
+
+James Bell
+CEO, Bellstone Engineering
+james@bellstone.ca
+
+* Review: Functions
+
+.play 10-methods/printPerson.go /BEGIN OMIT/,/END OMIT/
+
+* Review: User-defined types
+
+.play 10-methods/userType.go /BEGIN OMIT/,/END OMIT/
+
+* Method on a type
+
+.play 10-methods/methodPrintPerson.go /BEGIN OMIT/,/END OMIT/
+
+* Methods can change data
+
+.play 10-methods/incrementor.go /BEGIN OMIT/,/END OMIT/
+
+* Methods can change data by using pointers
+
+- In order to change data, the method must act on a pointer.
+- The only necessary change to the `Incrementor` is to make the `Increment` function take a pointer.
+
+.play 10-methods/incrementorFixed.go /BEGIN OMIT/,/END OMIT/
+
+* Go converts between pointers and values automatically
+
+- On the last slide, an `Incrementor` value was used to call a pointer method.
+- Here, the variable `inc` is created as a pointer instead of as a value, but there's no need to change the rest of the code.
+
+.play 10-methods/incrementorPointer.go /BEGIN OMIT/,/END OMIT/
+
+* Scope
+
+- As with functions and variables, capitalized methods can be accessed from other packages.
+- The incrementor's `int` is private, but all of the methods and the type are public.
```

Comment:
This isn't as easy to follow as it could be because `int` isn't shown on this slide.